packages/views/src/components/theme-provider.tsx

```diff
@@ -26,24 +26,40 @@ export function ThemeProvider({
   storageKey = 'canary-ui-theme',
   ...props
 }: ThemeProviderProps) {
+  const getMediaQuery = () => window.matchMedia('(prefers-color-scheme: dark)')
+
   const [theme, setThemeState] = useState<Theme>(() => (localStorage.getItem(storageKey) as Theme) || defaultTheme)
+  const [systemTheme, setSystemTheme] = useState<'dark' | 'light'>(() => (getMediaQuery().matches ? 'dark' : 'light'))
```

Comment:
maybe create an enum for themes?

Reply:
The system theme will always be either Dark or Light, it basically represents your currently selected laptop theme.

But, once we have multiple combinations for theme selection, we'll come up with a structured way to represent them.